dask/dataframe/multi.py

```diff
@@ -867,6 +886,14 @@ def merge_asof(
     result = merge_asof_indexed(left, right, **kwargs)
     if left_on or right_on:
         result = result.reset_index()
+        if ixcol is not None:
+            if divs is not None:
+                result = result.set_index(ixcol, sorted=True, divisions=divs)
+            else:
+                result = result.map_partitions(
+                    M.set_index, ixcol, meta=result._meta.set_index(ixcol)
+                )
```

Comment:
`meta` shouldn't be needed here, `map_partitions` should do the exact same thing automatically for you.

```suggestion
                result = result.map_partitions(M.set_index, ixcol)
```